org.eclipse.gef/src/org/eclipse/gef/ui/parts/TreeViewer.java

```diff
@@ -181,16 +182,25 @@ protected void fireSelectionChanged() {
 		showSelectionInTree();
 	}
 
+	/**
+	 * @since 3.16
+	 */
+	@Override
+	public Tree getControl() {
+		return (Tree) super.getControl();
+	}
```

Comment:
Perhaps it would be better to make the AbstractEditPartViewer generic with type \<T extends Control\>?

For the GraphicalViewer, T is a Canvas and for the TreeViewer, it's a Tree.

Reply:
Not sure. We only have two three viewers and with that style it is rather explicit that this viewer will always have a Tree.

Reply:
Fair enough. It would probably also require  changes to the EditPartViewer interface, so it's probably safer to keep it like this.